app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt

```diff
@@ -160,93 +160,92 @@ class ChatPresenter(chatActivity: ChatActivity) : IChatPresenter, IChatModel.OnR
     }
 
     override fun onRetrieveSuccess(response: Response<MemoryResponse>?) {
-        if (response != null && response.isSuccessful && response.body() != null) {
-            val allMessages = response.body()?.cognitionsList
-            if (allMessages != null) {
-                if (allMessages.isEmpty()) {
-                    chatView?.showToast("No messages found")
-                } else {
-                    var c: Long
-                    for (i in allMessages.size - 1 downTo 0) {
-                        val query = allMessages[i].query
-                        val queryDate = allMessages[i].queryDate
-                        val answerDate = allMessages[i].answerDate
-
-                        val urlList = ParseSusiResponseHelper.extractUrls(query)
-                        val isHavingLink = !urlList.isEmpty()
-
-                        newMessageIndex = PrefManager.getLong(Constant.MESSAGE_COUNT, 0)
-
-                        if (newMessageIndex == 0L) {
+        val res = response?.body()
```

Comment:
What is `res`?

Use expressive variable names, not abbreviations or short forms

Reply:
response is already in use.

Reply:
`response.body()` is not response, so it should not be named as response